### PR TITLE
Rework database schema loading

### PR DESCRIPTION
**Pull request by voievodin**
Adds new approach for initializing database schemas and performing migrations.
[Flyway](https://flywaydb.org/) with some custom modifications is used as database migration engine.
The issues are #2768 and #2110. 

### Configuration properties

```properties
# if database contains some data(schema is already initialized), 
# flyway will initialize its state without any migrations up to baseline version
db.schema.flyway.baseline.enabled=true
# up to this version <= migration scripts won't be applied on the existing database
db.schema.flyway.baseline.version=5.0.0.8.1
# migration scripts prefix(empty by default) e.g. it's "v" in "v1.init.sql" script name
db.schema.flyway.scripts.prefix=
# migration scripts suffixes(.sql)
db.schema.flyway.scripts.suffix=.sql
# the real script version is between prefix and first separator appearance
# e.g script version is "1" for the script with name "1.init.sql" according to given config
db.schema.flyway.scripts.version_separator=__
# where to search for scripts, it may be either classpath or filesystem
# if location starts with "classpath:" or with nothing then migrations resolver will try 
# to find scripts in classpath, otherwise if scripts starts with "filesystem:" then 
# it will lookup filesystem for scripts, the scripts must be structure in a way described below.
# It may contain several locations separated with ","
db.schema.flyway.scripts.locations=classpath:che-schema
# resource name to lookup data source
db.jndi.datasource.name=java:/comp/env/jdbc/che
```
### Scripts directory structure :file_folder:
Let's say we have 2 locations in classpath 

```
che-schema/
    5.0.0-M8/
        1__init.sql
    5.0.0-M9
        1__rename_tables.sql
        2__insert_predefined_data.sql
        postgresql/
            2__insert_predefined_data.sql
codenvy-schema/
    5.0.0-M8/
        1.1__init_codenvy.sql
    5.0.0-M9
        2.1__create_new_tables.sql
```
With the simplest configuration:
- prefix - empty
- suffix - `.sql`
- version separator - `__`
- locations - `classpath:che-schema,classpath:codenvy-schema`

If the current database driver is `posgresql` then the following migrations will be resolved and applied:

| db version        | script name                              | location                     |  picked vendor specific |
| -------------------- |:----------------------------------------:| --------------------------:|--------------------------------:|
|5.0.0.8.1            |  1__init.sql                                |  che-schema           | no                                   |
| 5.0.0.8.1.1        | 1.1__init_codenvy.sql               |  codenvy-schema   | no                                   |
| 5.0.0.9.1           | 1__rename_tables.sql              |  che-schema           | no                                   |
| 5.0.0.9.2           | 2__insert_predefined_data.sql |  che-schema           | yes                                  |
| 5.0.0.9.2.1        | 2.1__create_new_tables.sql     |  codenvy-schema   | no                                    |

@sleshchenko, @akorneta, @skabashnyuk please review the pull request

### CODE REVIEW

**Review by akorneta — APPROVED** [review]

File: core/che-core-db/src/main/java/org/eclipse/che/core/db/schema/impl/flyway/CustomSqlMigrationResolver.java [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+        final ListMultimap<String, SqlScript> scripts = finder.findScripts(flywayConfiguration);
+        LOG.debug("Found scripts: {}", scripts);
+
+        // filter sql scripts according to current db provider
+        final ListMultimap<String, SqlScript> versionToScripts = ArrayListMultimap.create();
+        for (String name : scripts.keySet()) {
+            final List<SqlScript> candidates = scripts.get(name);
+            final Map<String, SqlScript> vendorToScript = new HashMap<>();
+            for (SqlScript candidate : candidates) {
+                final String vendorName = candidate.vendor == null ? DEFAULT_VENDOR_NAME : candidate.vendor;
```
Comment (skabashnyuk) [inline core/che-core-db/src/main/java/org/eclipse/che/core/db/schema/impl/flyway/CustomSqlMigrationResolver.java]: firstNotNull?

File: core/che-core-db/src/test/resources/logback-test.xml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    </appender>
+
+    <appender name="file" class="ch.qos.logback.core.FileAppender">
+        <File>target/log/codenvy.log</File>
+        <encoder>
+            <pattern>%-41(%date[%.15thread]) %-45([%-5level] [%.30logger{30} %L]) - %msg%n</pattern>
+        </encoder>
+    </appender>
+
+    <!--logger name="com.codenvy.factory" level="DEBUG"/-->
```
Comment (sleshchenko) [inline core/che-core-db/src/test/resources/logback-test.xml]: let's remove this commented line and change log level from `INFO` to `ERROR`

File: pom.xml [hunk 3]
```diff
@@ -848,6 +860,16 @@
                 <version>${che.version}</version>
             </dependency>
             <dependency>
+                <groupId>org.flywaydb</groupId>
```
Comment (sleshchenko) [inline pom.xml]: I think you should move these dependencies to [che-dependencies](https://github.com/eclipse/che-dependencies/blob/master/pom.xml) project

Comment (voievodin) [reply pom.xml]: I will, thank you for the reminder

File: wsmaster/che-core-api-ssh/src/main/java/org/eclipse/che/api/ssh/server/model/impl/SshPairImpl.java [hunk 4]
```diff
@@ -43,20 +44,24 @@
         }
 )
 @IdClass(SshPairPrimaryKey.class)
+@Table(name = "sshkeypair")
 public class SshPairImpl implements SshPair {
     @Id
+    @Column(name = "owner")
```
Comment (sleshchenko) [inline wsmaster/che-core-api-ssh/src/main/java/org/eclipse/che/api/ssh/server/model/impl/SshPairImpl.java:50]: Looks like we should remove `user` field

**Review by sleshchenko — APPROVED** [review]

**Review by skabashnyuk — APPROVED** [review]

**Pull request merged**